internal/provider/resource_csrs.go

```diff
@@ -0,0 +1,241 @@
+				Description:  "ISO 3166 country code where company is legally registered",
+				Required:     true,
+				ValidateFunc: validation.StringInSlice(constants.GetIso3166CountryCodes(), false),
+			},
+			"email": {
+				Type:         schema.TypeString,
+				Description:  "Contact email address",
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+			"key_size": {
+				Type:         schema.TypeInt,
+				Description:  "Certificate public key size. One among: 2048, 3072, 4096",
+				Required:     true,
+				ValidateFunc: validation.IntInSlice([]int{2048, 3072, 4096}),
+			},
+			"locality": {
+				Type:         schema.TypeString,
+				Description:  "The city or locality where company is legally registered",
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+			"organization": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "The name under which your company is known. The listed organization must be the legal registrant of the domain name in the certificate request.",
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+			"organization_unit": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "Organization with which the certificate is associated",
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+			"state": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "Full name (do not abbreviate) of the state, province, region, or territory where your company is legally registered.",
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+			"resource": {
+				Type:     schema.TypeString,
+				Required: true,
+				// TODO when migrating to 5.x.x support check if these are still accurate
+				Description:  "Resources for which the CSRs are to be generated. One among: SDDC_MANAGER, VCENTER, NSX_MANAGER, NSXT_MANAGER, VROPS, VRSLCM, VXRAIL_MANAGER",
+				ValidateFunc: validation.StringInSlice([]string{"SDDC_MANAGER", "VCENTER", "NSX_MANAGER", "NSXT_MANAGER", "VROPS", "VRSLCM", "VXRAIL_MANAGER"}, false),
+			},
+			"csr": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "Resulting CSR",
+				Elem:        certificates.CsrSchema(),
+			},
+		},
+	}
+}
+
+func resourceCsrCreate(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcfClient := meta.(*api_client.SddcManagerClient)
+	apiClient := vcfClient.ApiClient
+
+	domainId := data.Get("domain_id").(string)
+	resourceType := data.Get("resource").(string)
+
+	resourceFqdn, err := getFqdnOfResourceTypeInDomain(ctx, domainId, resourceType, apiClient)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	if resourceFqdn == nil {
+		return diag.FromErr(fmt.Errorf("could not determine FQDN for resourceType %s in domain %s", resourceType, domainId))
+	}
+
+	country := data.Get("country").(string)
+	email := data.Get("email").(string)
+	keySize := strconv.Itoa(data.Get("key_size").(int))
```

Comment:
Wouldn't data.Get("key_size").(string) do the same ? Just speculating here.

Reply:
panic: interface conversion: interface {} is int, not string